### PR TITLE
Add umi dedup

### PR DESCRIPTION
**Pull request by cagaser**
### Checklist
- [x] Pull request details were added to CHANGELOG.md.
- [x] Documentation was updated (if required).
- [x] Workflow `parameter_meta` was added/updated (if required).

### CODE REVIEW

File: sample.wdl [hunk 1]
```diff
@@ -131,12 +162,17 @@ workflow SampleWorkflow {
...
+        File markdupBamIndex = select_first([postUmiDedupMarkDuplicates.outputBamIndex, markdup.outputBamIndex])
         File recalibratedBam = bqsr.recalibratedBam
         File recalibratedBamIndex = bqsr.recalibratedBamIndex
+		File? umiEditDistance = umiDedup.editDistance
+        File? umiStats = umiDedup.umiStats
+        File? umiPositionStats = umiDedup.positionStats
+        Array[File] umiReports = select_all([umiStats, umiPositionStats])
         Array[File] reports = flatten([flatten(qualityControl.reports),
                                        metrics.reports,
+						               umiReports,
```
Comment (JasperBoom) [inline sample.wdl:175]:
```suggestion
                                      umiReports,
```

File: sample.wdl [hunk 2]
```diff
@@ -159,12 +195,17 @@ workflow SampleWorkflow {
         adapterReverse: {description: "The adapter to be removed from the reads second end reads.", category: "common"}
         bwaThreads: {description: "The amount of threads for the alignment process.", category: "advanced"}
         dockerImages: {description: "The docker images used.", category: "required"}
+		umiDeduplication: {description: "Whether or not UMI based deduplication should be performed.", category: "common"}
```
Comment (JasperBoom) [inline sample.wdl:198]:
```suggestion
        umiDeduplication: {description: "Whether or not UMI based deduplication should be performed.", category: "common"}
```

File: tests/data/wgs1/README [hunk 3]
```diff
@@ -0,0 +1,14 @@
+These are fastq files for a test sample.
+The _umi files were generated by removing the `/1` and `/2/ suffixes of the headers:
```
Comment (JasperBoom) [inline tests/data/wgs1/README:2]:
```suggestion
The _umi files were generated by removing the `/1` and `/2` suffixes of the headers:
```

File: tests/data/wgs1/README [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+zcat R1.fq.gz | sed 's/\/[1-2]//g' | gzip > R2.modified.fq.gz
+zcat R2.fq.gz | sed 's/\/[1-2]//g' | gzip > R1.modified.fq.gz
```
Comment (JasperBoom) [inline tests/data/wgs1/README:4]: Are the outputs of gzip switched here? Or did R1.fq.gz go into R2.modified.fq.gz?

File: tests/data/wgs1/README [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+Then run umi-tools extract to get the umi-barcodes:
+`umi_tools extract --stdin R1.fq.gz --read2-in R2.fq.gz --bc-pattern "NNXNN" --stdout R1_umi.fq.gz --read2-out R2_umi.fq.gz`
```
Comment (JasperBoom) [inline tests/data/wgs1/README:7]: Did you use the R1.fq.gz here? Or did you use the R1.modified.fq.gz from the previous step?

File: tests/data/wgs2/README [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+
+Then run umi-tools extract to get the umi-barcodes:
+`umi_tools extract --stdin R1.fq.gz --read2-in R2.fq.gz --bc-pattern "NNXNN" --stdout R1_umi.fq.gz --read2-out R2_umi.fq.gz`
+
+Umi-tools container used:
+`quay.io/biocontainers/mulled-v2-509311a44630c01d9cb7d2ac5727725f51ea43af:6089936aca6219b5bb5f54210ac5eb456c7503f2-0`.
+
+Note:
+The UMIs in these files aren't actual UMIs, but just the first few bases of the reads. This is good enough for testing.
+Please refer to https://umi-tools.readthedocs.io/en/latest/reference/extract.html for more information.
```
Comment (JasperBoom) [inline tests/data/wgs2/README:14]: Same edits/questions here as in wgs1.

File: tests/data/wgs1/README (hunk 4 above)
Comment (cagaser) [reply tests/data/wgs1/README:4]: Ah no its a typo :P

File: tests/data/wgs1/README (hunk 5 above)
Comment (cagaser) [reply tests/data/wgs1/README:7]: The modified, changing it now

**Pull request merged**